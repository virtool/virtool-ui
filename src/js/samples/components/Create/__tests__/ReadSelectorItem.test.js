import { ReadSelectorItem, ReadOrientation } from "../ReadSelectorItem";

describe("<ReadOrientation />", () => {
    let props;
    beforeEach(() => {
        props = {
            index: 0,
            selected: true
        };
    });

    it("should render when [index=0]", () => {
        const wrapper = shallow(<ReadOrientation {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should render when [index=1]", () => {
        props.index = 1;
        const wrapper = shallow(<ReadOrientation {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should render when [selected=true]", () => {
        const wrapper = shallow(<ReadOrientation {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should render when [selected=false]", () => {
        props.selected = false;
        const wrapper = shallow(<ReadOrientation {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<ReadSelectorItem />", () => {
    let props;
    beforeEach(() => {
        props = {
            id: "foo",
            index: 0,
            name: "bar",
            selected: true,
            size: 0,
            onSelect: jest.fn()
        };
    });

    it("should render when [selected=true]", () => {
        const wrapper = shallow(<ReadSelectorItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should render when [selected=false]", () => {
        props.selected = false;
        const wrapper = shallow(<ReadSelectorItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should call onSelect when ReadSelectorItem is clicked", () => {
        const wrapper = shallow(<ReadSelectorItem {...props} />);
        wrapper.simulate("click");
        expect(props.onSelect).toHaveBeenCalledWith("foo");
    });
});
