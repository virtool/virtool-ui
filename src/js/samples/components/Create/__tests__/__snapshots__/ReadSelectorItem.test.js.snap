// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadOrientation /> should render when [index=0] 1`] = `
<ReadSelectorItem__StyledReadOrientation>
  LEFT
</ReadSelectorItem__StyledReadOrientation>
`;

exports[`<ReadOrientation /> should render when [index=1] 1`] = `
<ReadSelectorItem__StyledReadOrientation>
  RIGHT
</ReadSelectorItem__StyledReadOrientation>
`;

exports[`<ReadOrientation /> should render when [selected=false] 1`] = `""`;

exports[`<ReadOrientation /> should render when [selected=true] 1`] = `
<ReadSelectorItem__StyledReadOrientation>
  LEFT
</ReadSelectorItem__StyledReadOrientation>
`;

exports[`<ReadSelectorItem /> should render when [selected=false] 1`] = `
<ReadSelectorItem__StyledReadSelectorItem
  active={false}
  onClick={[Function]}
>
  <ReadSelectorItem__ReadTitle>
    <ReadSelectorItem__ReadIcon>
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="file"
      />
    </ReadSelectorItem__ReadIcon>
    <div>
      <strong>
        bar
      </strong>
      <div>
        0.0B
      </div>
    </div>
  </ReadSelectorItem__ReadTitle>
  <ReadOrientation
    index={0}
    selected={false}
  />
</ReadSelectorItem__StyledReadSelectorItem>
`;

exports[`<ReadSelectorItem /> should render when [selected=true] 1`] = `
<ReadSelectorItem__StyledReadSelectorItem
  active={true}
  onClick={[Function]}
>
  <ReadSelectorItem__ReadTitle>
    <ReadSelectorItem__ReadIcon>
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="file"
      />
    </ReadSelectorItem__ReadIcon>
    <div>
      <strong>
        bar
      </strong>
      <div>
        0.0B
      </div>
    </div>
  </ReadSelectorItem__ReadTitle>
  <ReadOrientation
    index={0}
    selected={true}
  />
</ReadSelectorItem__StyledReadSelectorItem>
`;
